src/basic/unit-name.c

```diff
@@ -507,6 +515,68 @@ int unit_name_template(const char *f, char **ret) {
         return 0;
 }
 
+bool unit_name_is_hashed(const char *name) {
+        char *s;
+
+        if (!unit_name_is_valid(name, UNIT_NAME_PLAIN))
+                return false;
+
+        assert_se(s = strrchr(name, '.'));
+
+        if (s - name < UNIT_NAME_HASH_LENGTH_CHARS + 1)
+                return false;
+
+        s -= UNIT_NAME_HASH_LENGTH_CHARS;
+        if (s[-1] != '_')
+                return false;
+
+        for (size_t i = 0; i < UNIT_NAME_HASH_LENGTH_CHARS; i++)
+                if (unhexchar(s[i]) < 0)
```

Comment:
hmm, here's something I just realized: unhexchar() actually allows both uppercase and lowercase chars. We should only allow lower-case here though. hence maybe:

```c
if (strchr(LOWERCASEHEXDIGITS, s[i]) < 0)
        …
```

And then define LOWERCASEHEXDIGITS near HEXDIGITS at the top of string-util.h.